modules/_labs/pagination/react/lib/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import {IconButtonProps} from '@workday/canvas-kit-react-button';
+
+import {PaginationModel} from './types';
+import {usePaginationModel, UsePaginationModelConfig} from './usePaginationModel';
+
+import {PaginationNav, PaginationNavProps} from './Nav';
+import {
+  JumpToFirstButton,
+  StepToPreviousButton,
+  StepToNextButton,
+  JumpToLastButton,
+  Controls,
+  ControlsProps,
+} from './Controls';
+
+import {PageList, PageListProps, PageListItem, PageListItemProps} from './PageList';
+import {PageButton, PageButtonProps} from './PageButton';
+
+import {GoTo} from './GoTo';
+import {GoToFormProps} from './GoTo/Form';
+import {GoToLabelProps} from './GoTo/Label';
+import {GoToTextInputProps} from './GoTo/TextInput';
+
+import {AdditionalDetails, AdditionalDetailsProps} from './AdditionalDetails';
+
+export const PaginationContext = React.createContext({} as PaginationModel);
+
+type PaginationConfigurationProps = PaginationNavProps & {
+  lastPage: number;
+  firstPage?: number;
+  initialCurrentPage?: number;
+  rangeSize?: number;
+  model?: never;
+  onPageChange?: (pageNumber: number) => void;
+};
+
+type PaginationModelProps = PaginationNavProps & {
+  lastPage?: never;
+  firstPage?: never;
+  initialCurrentPage?: never;
+  rangeSize?: never;
+  model: PaginationModel;
+  onPageChange?: never;
+};
+
+function useDefaultModel<T, C>(model: T | undefined, config: C, fn: (config: C) => T) {
+  return model || fn(config);
+}
+
+export type PaginationProps = PaginationConfigurationProps | PaginationModelProps;
+
+export const Pagination = (props: PaginationProps) => {
+  const model = useDefaultModel(props.model, props as UsePaginationModelConfig, usePaginationModel);
+
+  const {
+    'aria-label': ariaLabel,
+    children,
+    lastPage,
+    firstPage,
+    initialCurrentPage,
+    rangeSize,
+    onPageChange,
+    ...restProps
+  } = props;
+
+  return (
+    <PaginationContext.Provider value={model}>
+      <PaginationNav aria-label={ariaLabel} {...restProps}>
+        {children}
+      </PaginationNav>
+    </PaginationContext.Provider>
+  );
+};
+
+Pagination.Controls = (props: ControlsProps) => {
+  return <Controls {...props} />;
+};
+
+Pagination.JumpToFirstButton = (props: IconButtonProps) => {
+  // The linter doesn't recognize the dot syntax, so we're disabling the rule
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const model = React.useContext(PaginationContext);
+  return <JumpToFirstButton {...model} {...props} />;
+};
```

Comment:
This pattern ends up being a lot of  boilerplate. Tabs has context built into the sub-components to reduce this type of boilerplate. But these components aren't exactly using the Model API. This is doing `{...model}` instead of `model={model}`. Is this okay for now?

Reply:
> _This pattern ends up being a lot of boilerplate._

Very much agree. This will be resolved with `createComponent` in v5.

> _This is doing `{...model}` instead of `model={model}`_

Yep. This was before we solidified that pattern. I could update it now and that would (potentially) create less churn when users need to update to v5.

Reply:
Up to you. If we don't export `JumpToFirstButton`, I don't think the consumer will see much difference. It will show up in the React render tree as 2 different components. `ref` forwarding is not happening. If it was, we'd have to forward every time we create a new component wrapper. I don't think we're concerned about ref forwarding in v4 though.

Reply:
I switched all sub-components to use a model prop instead. 👍🏻